code/parachain/runtime/dali/src/migrations.rs

```diff
@@ -19,6 +21,616 @@ impl OnRuntimeUpgrade for SchedulerMigrationV1toV4 {
 	}
 }
 
+pub mod hard_coded_assets {
+
+	use super::*;
+	use assets_registry::WeightInfo;
+	use composable_traits::{
+		assets::{
+			AssetInfo, AssetInfoUpdate, BiBoundedAssetName, BiBoundedAssetSymbol,
+			InspectRegistryMetadata,
+		},
+		currency::{AssetExistentialDepositInspect, Rational64},
+		rational,
+	};
+	use frame_support::{
+		traits::{GetStorageVersion, StorageVersion},
+		WeakBoundedVec,
+	};
+
+	use primitives::topology;
+	use xcm::latest::prelude::*;
+
+	pub struct HardCodedAssetsMigration;
+
+	#[derive(Clone)]
+	struct AssetCreationInput {
+		asset_id: CurrencyId,
+		location: Option<XcmAssetLocation>,
+		asset_info: AssetInfo<Balance>,
+	}
+
+	impl AssetCreationInput {
+		fn new_asset(
```

Comment:
what benefit does this fn provide over just constructing `AssetCreationInput` by it's fields? IMO it could be removed, which would make the migration code easier to reason about instead of a function with 7 arguments

Reply:
imo it looks a little nicer when building a vector of assets for add_assets_to_storage fn. otherwise, i see what you mean.